src/PIL/CurImagePlugin.py

```diff
@@ -15,61 +15,209 @@
+from ._binary import o8
+from ._binary import o16le as o16
+from ._binary import o32le as o32
 
 #
 # --------------------------------------------------------------------
 
+_MAGIC = b"\x00\x00\x02\x00"
+
+
+def _save(im: Image.Image, fp: BytesIO, filename: str):
+    fp.write(_MAGIC)
+    bmp = im.encoderinfo.get("bitmap_format", "") == "bmp"
+    s = im.encoderinfo.get(
+        "sizes",
+        [(16, 16), (24, 24), (32, 32), (48, 48), (64, 64), (128, 128), (256, 256)],
+    )
+    h = im.encoderinfo.get("hotspots", [(0, 0) for i in range(len(s))])
+
+    if len(h) != len(s):
+        raise ValueError("Number of hotspots must be equal to number of cursor sizes")
+
+    # sort and remove duplicate sizes
+    sizes, hotspots = [], []
+    for size, hotspot in sorted(zip(s, h), key=lambda x: x[0]):
+        if size not in sizes:
+            sizes.append(size)
+            hotspots.append(hotspot)
+
+    frames = []
+    width, height = im.size
+    for size in sizes:
+        if size[0] > width or size[1] > height or size[0] > 256 or size[1] > 256:
+            continue
+
+        # TODO: invent a more convenient method for proportional scalings
+        frame = im.copy()
+        frame.thumbnail(size, Image.Resampling.LANCZOS, reducing_gap=None)
+        frames.append(frame)
+
+    fp.write(o16(len(frames)))  # idCount(2)
+    offset = fp.tell() + len(frames) * 16
+    for hotspot, frame in zip(hotspots, frames):
+        width, height = frame.size
+        # 0 means 256
+        fp.write(o8(width if width < 256 else 0))  # bWidth(1)
+        fp.write(o8(height if height < 256 else 0))  # bHeight(1)
+
+        bits, colors = BmpImagePlugin.SAVE[frame.mode][1:] if bmp else (32, 0)
+        fp.write(o8(colors))  # bColorCount(1)
+        fp.write(b"\0")  # bReserved(1)
+        fp.write(o16(hotspot[0]))  # x_hotspot(2)
+        fp.write(o16(hotspot[1]))  # y_hotspot(2)
+
+        image_io = BytesIO()
+        if bmp:
+            if bits != 32:
+                and_mask = Image.new("1", size)
+                ImageFile._save(
+                    and_mask, image_io, [("raw", (0, 0) + size, 0, ("1", 0, -1))]
+                )
+            else:
+                frame.alpha = True
+
+            frame.save(image_io, "dib")
+        else:
+            frame.save(image_io, "png")
+        image_io.seek(0)
+        image_bytes = image_io.read()
+        if bmp:
+            image_bytes = image_bytes[:8] + o32(height * 2) + image_bytes[12:]
+
+        bytes_len = len(image_bytes)
+        fp.write(o32(bytes_len))  # dwBytesInRes(4)
+        fp.write(o32(offset))  # dwImageOffset(4)
+        current = fp.tell()
+        fp.seek(offset)
+        fp.write(image_bytes)
+        offset = offset + bytes_len
+        fp.seek(current)
+
 
 def _accept(prefix):
-    return prefix[:4] == b"\0\0\2\0"
+    return prefix[:4] == _MAGIC
 
 
 ##
 # Image plugin for Windows Cursor files.
+class CurFile(IcoImagePlugin.IcoFile):
+    def __init__(self, buf: BytesIO):
```

Comment:
```suggestion
    def __init__(self, buf: BytesIO()):
```
Fixes https://github.com/python-pillow/Pillow/actions/runs/3138116114/jobs/5099677056#step:11:233
> /opt/hostedtoolcache/Python/3.10.7/x64/lib/python3.10/site-packages/PIL/CurImagePlugin.py:docstring of PIL.CurImagePlugin.CurFile:1: WARNING: py:class reference target not found: _io.BytesIO